### PR TITLE
feat: fulfil obligation tracking

### PR DESCRIPTION
**Pull request by vijayg10**

### CODE REVIEW

File: src/handlers/transfers/handler.js [hunk 1]
```diff
@@ -540,9 +540,15 @@ const processFulfilMessage = async (message, functionality, span) => {
           throw fspiopError
         }
       } else {
-        const payeeAccount = await Participant.getAccountByNameAndCurrency(transfer.payeeFsp, transfer.currency, Enum.Accounts.LedgerAccountType.POSITION)
-        await Kafka.proceed(Config.KAFKA_CONFIG, params, { consumerCommit, eventDetail, messageKey: payeeAccount.participantCurrencyId.toString(), topicNameOverride, hubName: Config.HUB_NAME })
-        histTimerEnd({ success: true, fspId: Config.INSTRUMENTATION_METRICS_LABELS.fspId })
+        const participantCurrencyId = await ProxyCache.deriveCurrencyId(transfer.payeeFsp, transfer.currency)
+        if (participantCurrencyId) {
+          await Kafka.proceed(Config.KAFKA_CONFIG, params, { consumerCommit, eventDetail, messageKey: participantCurrencyId.toString(), topicNameOverride, hubName: Config.HUB_NAME })
```
Comment (kleyow) [inline src/handlers/transfers/handler.js:545]: Would you not need a `checkSameCreditorDebtorProxy` check here when publishing the messageKey to see if the key needs to be `0`?

Comment (vijayg10) [reply src/handlers/transfers/handler.js]: You are right, I missed it. Its added now. Thanks

File: src/handlers/transfers/handler.js [hunk 2]
```diff
@@ -197,88 +198,90 @@ const processFulfilMessage = async (message, functionality, span) => {
...
-          ]
-        }
-      }
-      message.value.content.payload = reservedAbortedPayload
-      await Kafka.proceed(Config.KAFKA_CONFIG, params, { consumerCommit, eventDetail: reserveAbortedEventDetail, fromSwitch: true, toDestination: transfer.payeeFsp, hubName: Config.HUB_NAME })
-    }
-
-    throw apiFSPIOPError
   }
+  // Lets validate FSPIOP Source & Destination Headers
```
Comment (kleyow) [inline src/handlers/transfers/handler.js:202]: Why not use the headers to check if an fsp is inScheme and disabling these only if one is not in the scheme.
Feel like disabling these validatons entirely will break something in GP.

Comment (vijayg10) [reply src/handlers/transfers/handler.js]: Thanks @kleyow addressed your concern.

**Pull request merged**